version: "3.8"

services: 
  postgres:
    container_name: postgres_chainlink
    image: postgres:latest
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5
    ports:
      - 5432:5432
    volumes: 
      - db-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}

  ethereum:
    image: ethereum/client-go:v1.10.1
    ports:
      - 8546:8546
    command: --ropsten --syncmode light --ws --ipcdisable --ws.addr 0.0.0.0 --ws.origins="*" --datadir /geth
    volumes: 
      - geth:/geth

  chainlink:
    container_name: node_chainlink
    build: .
    env_file:
     .env
      # - ./.config/.chainlink-goerli/.env
      # - /.config/..chainlink-sepolia/.env      
    environment:
      - DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@host.docker.internal:5432/postgres?sslmode=disable
      - ETH_URL=wss://${ETH_URL}
    command: local node -p /chainlink/.password -a /chainlink/.api
    restart: on-failure
    ports: 
      - 6688:6688
    links:
      - postgres
    depends_on: 
      - postgres
    volumes:
      - chainlink_data:/chainlink

volumes: 
  geth:
  db-data:
  chainlink_data: